Source/Dafny/Dafny.atg

```diff
@@ -2579,46 +2579,78 @@ ExistentialGuard<out Expression e, bool allowLambda>
   Expression<out body, true, allowLambda>
   (. e = new ExistsExpr(x, bvars, null, body, attrs); .)
   .
-MatchStmt<out Statement/*!*/ s>
+
+
+/*------------------------------------------------------------------------*/
+ExtendedPattern<.out ExtendedPattern pat.>
+= (. IToken id; List<ExtendedPattern> arguments;
+    LiteralExpr lit; Expression te; BoundVar bv;
+    pat = null;
+
+  .)
+  ( IF(la.kind == _openparen)
+    "("                                   (. arguments = new List<ExtendedPattern>(); .)
+      [ ExtendedPattern<out pat>          (. arguments.Add(pat); .)
+      { "," ExtendedPattern<out pat>      (. arguments.Add(pat); .)
+      }]
+    ")"                                   (. id = t;
```

Comment:
Wouldn't it be better to use the token from the `(` instead of the token for the `)`?